### PR TITLE
Add Rowset and RowsetBuilder for PushHandler

### PR DESCRIPTION
**Pull request by kangpinghuang**
Remove SegmentGroup and ColumnDataWriter from PushHandler, replace them with Rowset and RowsetBuilder

### CODE REVIEW

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/push_handler.cpp [hunk 1]
```diff
@@ -73,13 +125,13 @@ OLAPStatus PushHandler::process_realtime_push(
 
         // if push finished, report success to fe
         if (tablet->has_pending_data(request.transaction_id)) {
```
Comment (chaoyli) [inline be/src/olap/push_handler.cpp]: Tablet will has no pending message. You should get it from transaction manager.

File: be/src/olap/push_handler.cpp [hunk 2]
```diff
@@ -54,10 +57,59 @@ OLAPStatus PushHandler::process_realtime_push(
...
     _request = request;
-    vector<TabletVars> tablet_infos(1);
-    tablet_infos[0].tablet = tablet;
-    AlterTabletType alter_tablet_type;
+    vector<TabletVars> tablet_vars(1);
+    tablet_vars[0].tablet = tablet;
+    res = _do_realtime_push(tablet, request, push_type, &tablet_vars, tablet_info_vec);
+    // if transaction existed in engine but push not finished, not report to fe
+    if (res == OLAP_ERR_PUSH_TRANSACTION_ALREADY_EXIST) {
+        OLAP_LOG_WARNING("find transaction existed when realtime push, not report. ",
```
Comment (chaoyli) [inline be/src/olap/push_handler.cpp]: LOG(WARNING)

File: be/src/olap/push_handler.cpp [hunk 3]
```diff
@@ -401,46 +394,42 @@ OLAPStatus PushHandler::_convert(
...
                 if (OLAP_SUCCESS != res) {
-                    OLAP_LOG_WARNING("read next row failed. [res=%d read_rows=%u]",
-                                     res, num_rows);
+                    LOG(WARNING) << "read next row failed."
+                                 << " res=" << res
+                                 << " read_rows=" << num_rows;
                     break;
                 } else {
-                    writer->next(row);
+                    if (OLAP_SUCCESS != (res = rowset_builder->add_row(&row))) {
```
Comment (chaoyli) [inline be/src/olap/push_handler.cpp]: AlphaRowsetBuilder add_row encapsulates attached_by function.
attached by function should be called previous by reader->next().

File: be/src/olap/push_handler.cpp [hunk 4]
```diff
@@ -454,13 +443,13 @@ OLAPStatus PushHandler::_convert(
...
-                    new_olap_indices);
+            // TODO(hkp): modify schema change
+            //SchemaChangeHandler schema_change;
+            //res = schema_change.schema_version_convert(
+            //        curr_tablet,
+            //        new_tablet,
+            //        curr_olap_indices,
+            //        new_olap_indices);
             if (res != OLAP_SUCCESS) {
                 OLAP_LOG_WARNING("failed to change schema version for delta."
```
Comment (chaoyli) [inline be/src/olap/push_handler.cpp]: LOG(WARNING)

File: be/src/olap/push_handler.h [hunk 5]
```diff
@@ -81,6 +76,15 @@ class PushHandler {
             const std::vector<TabletVars>& tablet_infos,
             std::vector<TTabletInfo>* tablet_info_vec);
 
+    OLAPStatus _do_realtime_push(
```
Comment (chaoyli) [inline be/src/olap/push_handler.h]: Using streaming_ingestion instead of realtime_push.

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/push_handler.cpp (hunk 3 above)
Comment (kangpinghuang) [reply be/src/olap/push_handler.cpp]: OK, I will fix it

**Review by chaoyli — APPROVED** [review]

**Pull request merged**